include/dice/template-library/flex_array.hpp

```diff
@@ -89,20 +196,22 @@ namespace dice::template_library {
 	 */
 	template<typename T, size_t extent_, size_t max_extent_ = extent_>
 	struct flex_array {
-		// extent_ != dynamic_extent -> extent_ == max_extent_
-		static_assert(extent_ == dynamic_extent || extent_ == max_extent_,
+		// extent_ == dynamic_extent -> max_extent_ != dynamic_extent
+		static_assert(extent_ != dynamic_extent || max_extent_ != dynamic_extent,
 					  "If extent is not dynamic_extent, extent must be equal to max_extent");
 
 		// extent_ == dynamic_extent -> max_extent_ != dynamic_extent
-		static_assert(extent_ != std::dynamic_extent || max_extent_ != std::dynamic_extent,
+		static_assert(extent_ != dynamic_extent || max_extent_ != dynamic_extent,
 					  "If extent is dynamic_extent, max_extent must not be dynamic_extent");
 
 		static constexpr size_t extent = extent_;
 		static constexpr size_t max_extent = max_extent_;
-		static constexpr bool is_dynamic_extent = extent == dynamic_extent;
+
+		static constexpr bool has_max_extent = max_extent != dynamic_extent;
+		static constexpr bool has_dynamic_extent = extent == dynamic_extent || max_extent == dynamic_extent;
```

Comment:
Also, stack_capacity would be nice to report. 
has_max_extend -> max_extend_
Svec -> not sure, probably https://github.com/martinus/svector/blob/74817cfbdbaceb415508de1ff10a0589bf8f25d7/include/ankerl/svector.h#L100

Reply:
Is in detail namespace I would rather not rely on that